playbooks/fundamentals/Container_Image_Build_Guidelines.adoc

```diff
@@ -0,0 +1,181 @@
+
+
+
+We normally start with an existing image that is stored in a registry like the one provided by Red Hat http://registry.access.redhat.com/[here].  The images stored here were also started from a “base” image.  Each image instantiated by adding to a new formed Dockerfile is another layer(s) added to the starting base image.  Let’s begin to build our very first image by starting with an example Dockerfile and get into the details as we go along.  Edit our file with an IDE or plain text editor like below:
+
+
+vim Dockerfile
+
+----
+FROM rhel7.5
+	MAINTAINER Matt Witz <mattwitz@example.com>
+----
+
+The *FROM* line is required to start from another image someone has created.
+
+The *MAINTAINER* line helps track changes to this new “base” image
+
+
+
+Now that we have these first two steps inside our file, we could run this container with a simple command of: docker build --rm -t rhel . (notice the . for the current working directory where our Dockerfile exists).  This image does not really do anything so let’s make it do something interesting like run a web server:
+
+
+
+
+----
+
+FROM rhel7.5
+	MAINTAINER Matt Witzmailto:mattwitz@example.com[ <mattwitz@example.c]om>
+LABEL version="1.0"
+	LABEL description="First image with Dockerfile."
+# Add Web server, update image, and clear cache
+	RUN yum -y install httpd && yum -y update; yum clean all
+	# Add some data to web server
+	RUN echo "This Web server is working." > /var/www/html/index.html
+	EXPOSE 80
+	ENTRYPOINT [ "/usr/sbin/httpd" ]
+	CMD [ "-D", "FOREGROUND" ]
+
+----
+
+
+
+We have added some very important lines to the file that we will now explain.  
+
+The *LABEL* line allows for some metadata that helps examine our images at a later date with some meaningful information with a command called docker inspect.
+
+_Note: Adding comments is a breeze just by prefacing the line with a # so others can easily understand our intent._
+
+The *RUN* line is our command execution within the image to add or remove or manipulate the images to our desired state.  We can have as many RUN statement lines but we suggest the minimum to get our container to run. 
+
+
+
+The *1st RUN* line executes package installs from one line separated by semicolons due to how a docker image is layered.  We will explain that in detail later in our guide and why that is important to consider.
+
+The *2nd RUN* line is adding some content to our web server.
+
+The *EXPOSE* line allows traffic to our web server on port 80 of the container.
+
+The *ENTRYPOINT* line is used to signify the command that will be run as if we were on the command line and this is the main execution process.
+
+The *CMD* line should be used to run the software contained by our image, along with any arguments. 
+
+
+
+== Onto the advanced use case and best practices
+
+==== Size does matter
```

Comment:
Don't skip header levels here. see the [style guide](https://github.com/redhat-cop/openshift-playbooks/blob/master/style_guide.adoc)